src/js/models/atom-node.js

```diff
@@ -0,0 +1,42 @@
+export default class AtomNode {
+  constructor(editor, atom, model, element, atomOptions) {
+    this.editor = editor;
+    this.atom = atom;
+    this.model = model;
+    this.atomOptions = atomOptions;
+    this.element = element;
+
+    this.setupResult = null;
+  }
+
+  render() {
+    this.teardown();
+
```

Comment:
Here we've discussing passing a document fragment instead. Pass it, then after `render` attach it to the `element`